sourced/ml/transformers/content2ids.py

```diff
@@ -0,0 +1,146 @@
+import gzip
+import operator
+import os
+import yaml
+
+import pygments
+from pygments.formatter import Formatter
+from pygments.lexers import get_lexer_by_name, ClassNotFound
+from pyspark import Row
+
+from sourced.ml.algorithms import TokenParser
+from sourced.ml.transformers import Transformer
+
+
+class Content2Ids(Transformer):
+
+    class FormatterProxy(Formatter):
+        name = "Proxy"
+        aliases = ["proxy"]
+        filenames = []
+
+        def __init__(self, **options):
+            super(Content2Ids.FormatterProxy, self).__init__(**options)
+            self.callback = options["callback"]
+
+        def format(self, tokensource, outfile):
+            self.callback(tokensource)
+
+    def __init__(self, args, documents_column, **kwargs):
+        super().__init__(**kwargs)
+        self.documents_column = documents_column
+        self.linguist2pygments = {}
+        self.split = args.split
+        self.idfreq = args.idfreq
+        self.output = args.output
+
+    def __call__(self, rows):
+        list_RDDs = []
+        self.build_mapping()
+        processed = rows.flatMap(self._process_row).persist()
```

Comment:
`persist()` may not be executed inside a Transformer unless it is a Cacher. Split the class into two and insert Cacher in the middle. Please note: in your case there should be only one Cacher for the whole pipeline tree.